src/drivers/CgreenDriver.cpp

```diff
@@ -0,0 +1,159 @@
+#include <cucumber-cpp/internal/drivers/CgreenDriver.hpp>
+
+#include <boost/bind.hpp>
+#include <boost/function.hpp>
+
+using namespace cgreen;
+
+namespace cucumber {
+namespace internal {
+
+
+bool CgreenStep::initialized = false;
+
+static const char* const blacklist[] = {
+        "Running",
+        "Completed",
+        "Failure",
+        "currentTest",
+        "\n",
+        "/"
+};
+
+class CukeCgreenInterceptor {
```

Comment:
This class smells: it's basically a singleton (the single `static` instance of this class) with every member being a singleton too (`static` member variables)! I suspect the quality of your destructor (which to me looks like it will trigger NULL pointer dereferences) is the result of this.

Please:

1. do without any singleton at all, or;
2. don't nest singletons and make the top-level one closer to the [singleton design pattern](https://en.wikipedia.org/wiki/Singleton_pattern) with lazy initialization (use a smart pointer for the instance pointer to ensure the destructor actually gets called).